src/js/components/uploadDetachedFiles/UploadDetachedFileMeta.jsx

```diff
@@ -0,0 +1,342 @@
+				.then((response) => {
+					this.setState({
+						detachedAward: item,
+						validationFinished: true,
+						headerErrors: false,
+						jobResults: response
+					});
+				});
+	}
+
+	parseJobStates(data) {
+		let runCheck = true;
+
+		if (data.jobs[0].job_status === 'failed' || data.jobs[0].job_status === 'invalid') {
+			// don't run the check again if it failed
+			runCheck = false;
+
+			let message = 'Error during D2 validation.';
+
+			if (!data.jobs[0].error_data[0] && data.jobs[0].error_data[0].error_description !== '') {
+				message = data.jobs[0].error_data[0].error_description;
+			}
+
+			// make a clone of the file's react state
+			const item = Object.assign({}, this.state.detachedAward);
+			item.status = "failed";
+
+			if(data.jobs[0].error_type === "header_errors") {
+				this.setState({
+					detachedAward: item,
+					validationFinished: true,
+					headerErrors: true
+				});
+			}
+			else {
+				this.validateSubmission(this.props.submission.id);
+			}
+		}
+		else if (data.jobs[0].job_status === 'finished') {
+			// don't run the check again if it's done
+			runCheck = false;
+
+			// display dowload buttons
+			// make a clone of the file's react state
+			const item = Object.assign({}, this.state.detachedAward);
+			item.status = "done";
+
+			this.validateSubmission(this.props.submission.id);
+		}
+
+		if (this.isUnmounted) {
+			return;
+		}
+
+		if (runCheck && !this.isUnmounted) {
+			// wait 5 seconds and check the file status again
+			window.setTimeout(() => {
+				this.checkFileStatus(this.props.submission.id);
+			}, timerDuration * 1000);
+		}
+	}
+
+	// ERRORS
+	// 1: Submission is already published
+	// 2: Fetching file metadata failed
+	// 3: File already has been submitted in another submission
+
+	render() {
+		let subTierAgencyClass = '';
+		let datePicker = null;
+		let uploadFilesBox = null;
+
+		if (this.state.agencyError) {
+			subTierAgencyClass = ' error usa-da-form-icon';
+		}
+
+		if (this.state.showDatePicker) {
+			let value = {
+				datePlaceholder: "Action",
+				endDate: this.state.detachedAward.endDate, 
+				label: "File D2: Financial Assistance",
+				startDate: this.state.detachedAward.startDate
+			}
+			datePicker = <DateRangeWrapper {...this.state} 
+								handleDateChange={this.handleDateChange.bind(this, "detachedAward")} 
+								updateError={this.updateError.bind(this)} 
+								value={value}
+								disabled={this.state.showValidationBox} />;
+		}
+
+		
+		if (this.state.detachedAward.valid && this.state.showUploadFilesBox) {
+			uploadFilesBox = <UploadDetachedFilesBox {...this.state} 
+								submission={this.props.submission}
+								uploadFile={this.uploadFile.bind(this)} />;  
+		}
+
+		let errorMessage = null;
+
+		if (this.state.detachedAward.error.show) {
+			errorMessage = <UploadDetachedFilesError error={this.state.detachedAward.error} />
+		}
+
+		let subtierAgency = null;
+
+		if(!this.state.showValidationBox){
```

Comment:
Everything to do with `this.state.showValidationBox` is probably irrelevant now because that's a separate page now isn't it?